docs/WOFpyODM2LBR_installation.md

```diff
@@ -235,7 +235,9 @@ cd $HOME/wofpyserverdev
 3. Edit `singlerunserver.py` to use the mysql config file.
 
 	```python
-	dao = Odm2Dao(get_connection('odm2_config_mysql.cfg'))
+	parser.add_argument('--config',
```

Comment:
This is basically doing the same thing. Not sure what the difference is. See: https://github.com/ODM2/WOFpy/blob/master/wof/examples/flask/odm2/timeseries/runserver_odm2_timeseries.py#L42

Reply:
I think maybe a change to specify that one needs to put in the correct path to the `.cfg` would work better.

Reply:
@lsetiawan, FYI, I won't chime in on this unless you ask me to :wink: It would take me a good chunk of digging to be able to say anything intelligent about this. You're the guru here!

Reply:
@lsetiawan the difference is for me when I did this
 dao = Odm2Dao(get_connection('odm2_config_postgresql.cfg'))

this line of code failed with an error but when I added it as the default in the parser argument it worked

Reply:
@miguelcleon 

```python
dao = Odm2Dao(get_connection(args.config))
``` 
is basically using the value from `args.config`, which in your change, if the user doesn't specify `--config` will be  `'odm2_config_mysql.cfg'`. So this means that `args.config` is `'odm2_config_mysql.cfg'`. So the runserver script ends up running, which is my original documentation:

```python
dao = Odm2Dao(get_connection('odm2_config_mysql.cfg'))
``` 

The breakage comes if `'odm2_config_mysql.cfg'` is not in the same folder as the runserver script. So I think a change of maybe adding `os.path.join(config_folder, 'odm2_config_mysql.cfg')` is maybe better and explain that with that, one has to know where the config file lives. or if same folder, one can use `os.curdir`.

Makes sense? Let me know if that's confusing.

@emiliom  do you think changing the default to 'odm2_config_mysql.cfg' is better than straight up changing the function that actually take the value? Thanks.

Reply:
@miguelcleon Our message crossed. Hmm.. that's weird. I tried both ways before, and both worked. So not sure why it doesn't work. Because if you look at [L25](https://github.com/ODM2/WOFpy/blob/master/wof/examples/flask/odm2/timeseries/runserver_odm2_timeseries.py#L25). The script just opens that file, and if you don't specify the path, it'll just open the file directly within the same folder..

```python
def get_connection(conf):
    """Get connection string from .cfg file.
    :param conf: ODM2 Config File. Ex. 'odm2_config_timeseries.cfg'
    :return: Connection String
    """
    config = configparser.ConfigParser()
    with open(conf, 'r') as configfile:
        config.read_file(configfile)
        connection = config['Database']['Connection_String']

return connection
```

Reply:
I get this error: 
```
[Fri Oct 20 18:54:26.046338 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843] mod_wsgi (pid=853): Exception occurred processing WSGI script '/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/wsgi.py'.
[Fri Oct 20 18:54:26.046351 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843] Traceback (most recent call last):
[Fri Oct 20 18:54:26.046362 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/wsgi.py", line 3, in <module>
[Fri Oct 20 18:54:26.046394 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843]     from runserver import app as application
[Fri Oct 20 18:54:26.046433 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/runserver.py", line 43
[Fri Oct 20 18:54:26.046437 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843]     dao = Odm2Dao(get_connection('/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/odm2_config_timeseries.cfg')))
[Fri Oct 20 18:54:26.046440 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843]                                                                                                                         ^
[Fri Oct 20 18:54:26.046443 2017] [wsgi:error] [pid 853:tid 140228907034368] [remote 165.123.20.27:50843] SyntaxError: invalid syntax
```

Reply:
actually this error, looks like I introduced another error in that last one (an extra ')')
```
[Fri Oct 20 18:57:12.668832 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843] mod_wsgi (pid=1467): Exception occurred processing WSGI script '/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/wsgi.py'.
[Fri Oct 20 18:57:12.668848 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843] Traceback (most recent call last):
[Fri Oct 20 18:57:12.668860 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/wsgi.py", line 3, in <module>
[Fri Oct 20 18:57:12.668881 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     from runserver import app as application
[Fri Oct 20 18:57:12.668888 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/runserver.py", line 45, in <module>
[Fri Oct 20 18:57:12.668909 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     app = wof.flask.create_wof_flask_app(dao, args.config)
[Fri Oct 20 18:57:12.668915 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/wof/flask/__init__.py", line 198, in create_wof_flask_app
[Fri Oct 20 18:57:12.668987 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     app = create_wof_flask_multiple({wConf},templates=templates)
[Fri Oct 20 18:57:12.668995 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/wof/flask/__init__.py", line 209, in create_wof_flask_multiple
[Fri Oct 20 18:57:12.669005 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     wof_obj_1_0 = wof_1_0.WOF(wConf.dao, wConf.config, templates)
[Fri Oct 20 18:57:12.669010 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/wof/core_1_0.py", line 38, in __init__
[Fri Oct 20 18:57:12.669182 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     self.config_from_file(config_file)
[Fri Oct 20 18:57:12.669190 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/wof/core_1_0.py", line 44, in config_from_file
[Fri Oct 20 18:57:12.669201 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     config = core.WOFConfig(file_name)
[Fri Oct 20 18:57:12.669206 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/wof/core.py", line 114, in __init__
[Fri Oct 20 18:57:12.669215 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     self.network = config.get('WOF', 'Network')
[Fri Oct 20 18:57:12.669220 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/backports/configparser/__init__.py", line 792, in get
[Fri Oct 20 18:57:12.671711 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     d = self._unify_values(section, vars)
[Fri Oct 20 18:57:12.671726 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/miniconda2/envs/wofpy/lib/python2.7/site-packages/backports/configparser/__init__.py", line 1162, in _unify_values
[Fri Oct 20 18:57:12.671742 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     raise NoSectionError(section)
[Fri Oct 20 18:57:12.671758 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843] NoSectionError: No section: 'WOF'
```

Reply:
@lsetiawan yeah, I did include the full path in my actual edited code. I see what your saying, but yeah for some reason I get an error if I do it the other way.

Reply:
I see the problem:

```
File "/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/wsgi.py", line 3, in <module>
[Fri Oct 20 18:57:12.668888 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]   File "/home/azureadmin/webapps/wofpyserverprod/odm2/timeseries/runserver.py", line 45, in <module>
[Fri Oct 20 18:57:12.668909 2017] [wsgi:error] [pid 1467:tid 140228907034368] [remote 165.123.20.27:50843]     app = wof.flask.create_wof_flask_app(dao, args.config)
```

The file name needs to be specified here too:
```
app = wof.flask.create_wof_flask_app(dao, args.config)
```

The runserver is pulling from the

Reply:
It's the no section 'WOF' error, which didn't seem to make sense.

Reply:
because `args.config` is `None` on the line I listed above

Reply:
so the better change would be:

```python
configfile = os.path.join(cfgpath, 'odm2_config_timeseries.cfg')
dao = Odm2Dao(get_connection(configfile))
app = wof.flask.create_wof_flask_app(dao, configfile)
```

Reply:
@lsetiawan yes, that is it 😄

Reply:
@lsetiawan Seems kind of better to change it in one place rather then two?

Reply:
That's fine, but you won't be able to show that it should be from absolute path. Maybe I'll change that in the actual runserver script. Thanks!